sfge/src/main/java/com/salesforce/rules/PathBasedRuleRunner.java

```diff
@@ -58,6 +58,14 @@ public PathBasedRuleRunner(
      * @return a set of violations that were detected
      */
     public Set<Violation> runRules() {
+        if (allRules.isEmpty()) {
```

Comment:
Exit early if no rules are provided. This should technically be impossible, but it's still worth having.
Question: Do we want to have a (possibly verbose-only) message logged to the user if this actually happens?

Reply:
Thanks for handling this. Since we're trying to move towards surfacing the config to run Graph Engine, invocation without rules could be a possibility. I like your idea of putting a warning and we probably want to show them always. Having a real-time message might be the cleanest.

Reply:
In the interest of time, I'd like to just add a TODO here and create an associated work item. Acceptable?